volmdlr/faces.py

```diff
@@ -6466,78 +6466,86 @@ def rotation_inplace(self, center: volmdlr.Point3D, axis: volmdlr.Vector3D,
-                y = min(ymin + j * resolution, ymax)
-                if y == ymin:
-                    y = ymin + 0.01 * resolution
-                points_box.append(volmdlr.Point2D(x, y))
-
-        points = [pt.copy() for pt in self.points]
-        for pt in points_box:
-            if t_poly2d.point_belongs(pt):
-                points.append(pt.to_3d(frame.origin, frame.u, frame.v))
-            elif t_poly2d.point_over_contour(pt):
-                points.append(pt.to_3d(frame.origin, frame.u, frame.v))
-
-        return volmdlr.Vector3D.remove_duplicate(points)
+
+        lengths = [self.points[0].point_distance(self.points[1]),
+                   self.points[1].point_distance(self.points[2]),
+                   self.points[2].point_distance(self.points[0])]
+        ls_max = max(lengths)
+
+        if ls_max <= resolution:
+            return self.points
+
+        pos_ls_max = lengths.index(ls_max)
+        p0, p1, p2 = self.points[-3 + pos_ls_max], self.points[-3 + pos_ls_max + 1], self.points[-3 + pos_ls_max + 2]
+
+        v_init = p0 - p1
+        v_init.normalize()
+        p0_res = []
+
+        nb = int(ls_max / resolution) + 2
+        for k in range(nb):
+            if k == 0:
+                p0_res.append(p1)
+            l = min(k * resolution, ls_max)
+            p0_res.append(p1 + v_init * l)
+
+        v1, l1 = p2 - p1, p2.point_distance(p1)
+        v1.normalize()
+        p1_res = []
+
+        for k, p0_k in enumerate(p0_res):
+            if k == 0:
+                point_v1 = p1
+            l = min(p0_res[0].point_distance(p0_k) * l1 / ls_max, l1)
+            point_v1 = p1 + v1 * l
+
+            l_int = point_v1.point_distance(p0_k)
+            nb_int = int(l_int / resolution) + 2
+            if nb_int == 2:
+                p1_res.append(point_v1)
+            else:
+                v_int = point_v1 - p0_k
+                v_int.normalize()
+                step_in = l_int / (nb_int - 1)
+                for i in range(nb_int):
+                    l = min(i * step_in, l_int)
+                    if l != 0:
+                        p1_res.append(p0_k + v_int * l)
+
+        return npy.unique(p0_res + p1_res).tolist()
 
     def subdescription_to_triangles(self, resolution=0.01):
         """
         Returns a list of Triangle3D with resolution as max
         length of subtriangles side.
         """
 
-        frame = self.surface3d.frame
-        pts2d = [pt.to_2d(frame.origin, frame.u, frame.v) for pt in self.points]
+        sub_triangles, done = [self.points], False
 
-        t_poly2d = volmdlr.wires.ClosedPolygon2D(pts2d)
-
-        sub_triangles2d = [t_poly2d]
-        done = False
         while not done:
-            triangles2d = []
-            for t, subtri in enumerate(sub_triangles2d):
-                ls_length = [ls.length() for ls in subtri.line_segments]
-                ls_max = max(ls_length)
+            triangles = []
+            for t, subtri in enumerate(sub_triangles):
```

Comment:
It seems t is no being used. If it is the case, could delete it along side with the enumerate?

Reply:
Thanks for the remark, I delete it